src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -171,15 +176,16 @@ public async Task<IRemoteProjectFile> LoadProjectFileAsync(string projectFilePat
         };
 
         _logger.LogInformation($"Loading {projectFilePath}");
-        return new RemoteProjectFile(await projectLoader.LoadProjectFileAsync(projectFilePath, _buildManager, cancellationToken).ConfigureAwait(false));
+        var projectFile = await projectLoader.LoadProjectFileAsync(projectFilePath, _buildManager, CancellationToken.None).ConfigureAwait(false);
+        return _server.AddTarget(projectFile);
     }
 
-    public Task<string?> TryGetProjectOutputPathAsync(string projectFilePath, CancellationToken cancellationToken)
+    public Task<string?> TryGetProjectOutputPathAsync(string projectFilePath)
     {
         EnsureMSBuildLoaded(projectFilePath);
         CreateBuildManager();
 
-        return _buildManager.TryGetOutputFilePathAsync(projectFilePath, cancellationToken);
+        return _buildManager.TryGetOutputFilePathAsync(projectFilePath, CancellationToken.None);
```

Comment:
i am not a fan.  i would still prefer these take cancellation tokens, and we thread things outwards as much as possible.

Reply:
Change was made to pass tokens around (and just the RPC layer is sticking in the CancellationToken.None)